### PR TITLE
add lua strings entry

### PR DESCRIPTION
**Pull request by qiqi0211**
### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/strings/strings.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+Title: 'Strings' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'A **string** is a sequence of characters of any length that can include letters, numbers, symbols, and spaces.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:3]:
```suggestion
Description: 'Strings are a sequence of characters of any length that can include letters, numbers, symbols, and spaces.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
```

We can't put markdown styling in the description.

File: content/lua/concepts/strings/strings.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-lua'
+---
+
+## Syntax
+
+In Lua, a string is a sequence of characters surrounded by a pair of single quotes `''` or double quotes `""`.
+
+```lua
+location = “Central Park” -- “Central Park” is a String
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:17]:
```suggestion
location = “Central Park” -- “Central Park” is a string
```
doesn't need to be capitalized

File: content/lua/concepts/strings/strings.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+## Concatenation
+
+Multiple variables can be combined together using the concatenation operator `(..)`.
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:22]:
```suggestion
Multiple variables can be combined together using the concatenation operator `..`.
```

I'd remove the parenthesis `()` so that people don't think that's part of the operator

File: content/lua/concepts/strings/strings.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+```lua
+time = 10
+location = “Central Park”
+print(“We will arrive at “ .. location .. “ at “.. time)
+```
+
+## Escape Characters
+
+Escape characters are used for clarity and conciseness. Sometimes Lua may interpret a character in a string, which can cause errors. To avoid this, escape characters can be used to clarify that a character should only be part of the string.
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:32]:
```suggestion
Escape characters are used for clarity and conciseness. Sometimes, Lua may interpret a character in a string, which can cause errors. To avoid this, escape characters can be used to clarify that a character should only be part of the string.
```

File: content/lua/concepts/strings/strings.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:33]: Suggest adding in an example to demonstrate this. Maybe something along these lines:

```lua
brokenString = "They said, "Hello!"" -- This will cause "Hello!" to be outside of the string

revisedString = "They said, \"Hello!\"" -- This will keep "Hello!" inside of the string using escape characters
```

File: content/lua/concepts/strings/strings.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+|vertical tab|\v|
+|backslash|\ \ |
+|quotation mark [double quote]|\ "|
+|apostrophe [single quote]|\ '|
+
+## String Functions
+
+ Lua's String Library contains many different string functions that can be used to manipulate strings. Below are some examples of useful functions.
+
+`string.len()`
```
Comment (lucwhite) [inline content/lua/concepts/strings/strings.md:53]: These should likely be terms, but that is outside of the scope of the internship. Make note of this in the handoff document

**Review by lucwhite — CHANGES_REQUESTED** [review]
Great work! Just some small changes 👍

File: content/lua/concepts/strings/strings.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Strings' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'Strings are a sequence of characters of any length that can include letters, numbers, symbols, and spaces.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Strings'
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:7]:
```suggestion
  - 'Strings'
  - 'Characters'
```

File: content/lua/concepts/strings/strings.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+__Strings__ are a sequence of characters of any length that can include letters, numbers, symbols, and spaces.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:13]:
```suggestion
**Strings** are a sequence of characters of any length that can include letters, numbers, symbols, and spaces.
```

File: content/lua/concepts/strings/strings.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+In Lua, a string is a sequence of characters surrounded by a pair of single quotes `''` or double quotes `""`.
+
+```lua
+location = “Central Park” -- “Central Park” is a string
+```
+
+## Concatenation
+
+Multiple variables can be combined together using the concatenation operator `..`.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:25]:
```suggestion
Multiple strings can be combined together using the concatenation operator `..`.
```

File: content/lua/concepts/strings/strings.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+```lua
+time = 10
+location = “Central Park”
+print(“We will arrive at “ .. location .. “ at “.. time)
+```
+
+## Escape Characters
+
+Escape characters are used for clarity and conciseness. Sometimes, Lua may interpret a character in a string, which can cause errors. To avoid this, escape characters can be used to clarify that a character should only be part of the string.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:35]:
```suggestion
Escape characters are used for clarity and conciseness. Sometimes, Lua may interpret a character in a string, which can cause errors. To avoid this, escape characters can be used to clarify that a character should be read as a string.
```

File: content/lua/concepts/strings/strings.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+brokenString = "They said, "Hello!"" -- This will cause "Hello!" to be outside of the string
+
+revisedString = "They said, \"Hello!\"" -- This will keep "Hello!" inside of the string using escape characters
+```
+
+The following is a list of some useful escape characters supported by Lua strings:
+
+| Description | Escape Characters|
+| -- | --|  
+|bell | \a |
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:47]:
```suggestion
|bell | `\a` |
```

Comment (caupolicandiaz) [reply content/lua/concepts/strings/strings.md:47]: repeat formatting for the other rows

File: content/lua/concepts/strings/strings.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+Return the length of the input string.
+
+`string.lower()`
+Return a copy of the input string with all uppercase letters changed to lowercase.  
+
+`string.upper()`
+Return a copy of the input string with all lowercase letters changed to uppercase.
+
+`string.reverse()`
+Return a string that is the input string reversed.
```
Comment (caupolicandiaz) [inline content/lua/concepts/strings/strings.md:72]: these line can be removed. eventually we'll add a terms folder with entries for each

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@qiqi0211 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks
